src/components/pages/pages.pw.tsx

```diff
@@ -0,0 +1,218 @@
+    test(`should render component with transition set to ${transition}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PagesComponent transition={transition} />);
+
+      const pageComponent = getDataElementByValue(page, "visible-page").nth(0);
+
+      await dataComponentButtonByText(page, "Open Preview").click();
+      await dataComponentButtonByText(page, "Go to second page").click();
+      const classAttribute = await pageComponent.getAttribute("class");
+
+      await expect(classAttribute).toContain(transition);
+    });
+  });
+
+  test("should render component and go next to Second page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PagesComponent />);
+
+    await dataComponentButtonByText(page, "Open Preview").click();
+    await dataComponentButtonByText(page, "Go to second page").click();
+
+    await expect(getDataElementByValue(page, "title").nth(0)).toHaveText(
+      "My Second Page"
+    );
+  });
+
+  test("should render component and go next to Third page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PagesComponent initialPageIndex={1} />);
+
+    await dataComponentButtonByText(page, "Open Preview").click();
+    await dataComponentButtonByText(page, "Go to third page").click();
+
+    await expect(getDataElementByValue(page, "title").nth(0)).toHaveText(
+      "My Third Page"
+    );
+  });
+
+  test("should render component and go back to Second page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PagesComponent initialPageIndex={2} />);
+
+    await dataComponentButtonByText(page, "Open Preview").click();
+    await backArrow(page).click();
+
+    await expect(getDataElementByValue(page, "title").nth(0)).toHaveText(
+      "My Second Page"
+    );
+  });
+
+  test("should render component and go back to First page", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PagesComponent initialPageIndex={1} />);
+
+    await dataComponentButtonByText(page, "Open Preview").click();
+    await backArrow(page).click();
+
+    await expect(getDataElementByValue(page, "title").nth(0)).toHaveText(
+      "My First Page"
+    );
+  });
+
+  [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS].forEach(
+    ([testTitle]) => {
+      test(`Page component should render as expected with title prop set to ${testTitle}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<PageComponentWithTitle testTitle={testTitle} />);
+
+        const pageComponent = getComponent(page, "page");
+        await waitForAnimationEnd(pageComponent);
+
+        await expect(pageComponent).toHaveText(`${testTitle}`);
+      });
+    }
+  );
+});
+
+test.describe("Accessibility tests for Pages component", () => {
+  [
+    ["number", 1],
+    ["string", "1"],
+  ].forEach(([type, propValue]) => {
+    test(`When initialPageIndex prop is passed as ${type}, should show the title that matches the one on the page with that index`, async ({
```

Comment:
Can we change the test description to include "check accessibility" which is the main purpose for the test. 
Same as below test